### DISCUSSION TITLE
Keep Block Array Data - Mesh Operations With --mesh-only

### DISCUSSION THREAD

**Discussion by Florian-Konrad** [idaholab/moose#25319] [Q&A Meshing] [answered]
Hi there :)

I'm running some mesh manipulations (eg refinement) on an exodus file containing material parametrization as block arrays. I'm doing that with an input file containing only the mesh block together with --mesh-only.

I prefer to do the mesh stuff in a separate step and would still like to be able use the block array information in the subsquent simulations (input files that only use FileMeshGenerator, loading the manipulated and checked mesh file).

Is there a way to keep element data when using --mesh-only? I tried use_for_exodus_restart = true with my FileMeshGenerator without any effect.

Cheers
Florian

**Comment by GiudGiud** [comment]
Hello

Yes in the filemeshgenerator you use this parameter:
exodus_extra_element_integers 
to keep the element integers that are already defined on the mesh

Guillaume

**Reply by Florian-Konrad** [reply]
Awesome, works! I totally overlooked this.  Thank you!

**Reply by Florian-Konrad** [reply]
oh I just noticed that this is only working with integers as the name of the parameter suggests. My material values eg are rock permeability which are floats like 1e-14 and this gets rounded to zero. Any way to preserve these values?

**Reply by GiudGiud** [reply]
Hello

Yes! That s a different thing. In your file_mesh_generator add use_for_exodus_restart then in your auxvariables, you specify the initial_from_file_var field. 
You can also specify the step

Guillaume

**Reply by Florian-Konrad** [reply]
this is possible while using --mesh-only flag?

**Reply by GiudGiud** [reply, marked as answer]
nope

**Reply by Florian-Konrad** [reply]
alright thanks! :)
